src/device/device-management.ts

```diff
@@ -1,66 +1,66 @@
 import * as media from '../media';
+import { MicrophoneConstraints } from '../media/local-audio-track';
 import { LocalCameraTrack } from '../media/local-camera-track';
 import { LocalDisplayTrack } from '../media/local-display-track';
 import { LocalMicrophoneTrack } from '../media/local-microphone-track';
+import { CameraConstraints, DisplayConstraints } from '../media/local-video-track';
 
-export enum ErrorTypes {
-  DEVICE_PERMISSION_DENIED = 'DEVICE_PERMISSION_DENIED',
-  CREATE_CAMERA_TRACK_FAILED = 'CREATE_CAMERA_TRACK_FAILED',
-  CREATE_MICROPHONE_TRACK_FAILED = 'CREATE_MICROPHONE_TRACK_FAILED',
-}
+import { ErrorTypes, WebrtcError } from '../error/WebrtcError';
+import { LocalComputerAudioTrack } from '../media/local-computer-audio-track';
 
 /**
- * Represents a WCME error, which contains error type and error message.
+ * Creates a camera video track. Please note that the constraint params in second getUserMedia call would NOT take effect when:
+ *
+ * 1. Previous captured video track from the same device is not stopped .
+ * 2. Previous createCameraTrack() call for the same device is in progress.
+ 
+ * Creates LocalMicrophoneTrack and LocalCameraTrack at the same time.
+ *
+ * @param audioConstraints - Audio constraints to create microphone track.
+ * @param videoConstraints - Video constraints to create camera track.
+ * @returns LocalMicrophoneTrack and LocalCameraTrack at same time.
  */
-export class WcmeError {
-  type: string;
-
-  message: string;
-
-  /**
-   * Creates new error.
-   *
-   * @param type - Error type.
-   * @param message - Error message.
-   */
-  constructor(type: ErrorTypes, message = '') {
-    this.type = type;
-    this.message = message;
+export async function createMicrophoneAndCameraTracks(
+  audioConstraints?: MicrophoneConstraints,
```

Comment:
I just noticed: should these support `boolean`s as well? i.e.

```
audioConstraints?: MicrophoneConstraints | boolean
```

Reply:
done , can do it for video also , it will look little odd as there is already a function for micropone and camera